kevm-pyk/src/tests/integration/test_prove.py

```diff
@@ -253,6 +282,7 @@ def test_pyk_prove(
             definition_dir=target.definition_dir,
             includes=[str(include_dir) for include_dir in config.INCLUDE_DIRS] + target.includes,
             save_directory=use_directory,
+            max_depth=300,  # Workaround for ecrecover00-siginvalid issue
```

Comment:
As it turned out, test `ecrecover00-siginvalid-spec.k` suffers from a growing heap when using booster.

This change reduces the maximum depth for the prover tests to a value that works - after each 300 steps of execution, the configuration and constraints is simplified before returning to `pyk`.
This is not a permanent limitation but rather a workaround that unblocks dependent development while the root cause for the regression is investigated and addressed.

Reply:
Have we made an issue to track the memory leak in haskell backend for this one? Maybe we can link it here?

Reply:
Tracked in this issue: https://github.com/runtimeverification/hs-backend-booster/issues/347